### ISSUE TITLE
Validation failed

### ISSUE THREAD

**Issue by hanskoff** [gocardless/gocardless-pro-java#6]
Hi,
When I try to create bank account for GB with sort code in 6-digit format (dddddd) I get exception (it would be nice to see exactly which parameter fails validation btw. not only stacktrace) but it seems that gocardless api accepts `dd-dd-dd` format only  because I can register bank account in that case which is not what one might expect reading this https://developer.gocardless.com/pro/2015-07-06/#appendix-local-bank-details, or am I doing something wrong?
```
com.gocardless.errors.ValidationFailedException: Validation failed
	at com.gocardless.errors.GoCardlessErrorMapper.toException(GoCardlessErrorMapper.java:22) ~[app.jar:0.0.1-SNAPSHOT]
	at com.gocardless.http.ResponseParser.parseError(ResponseParser.java:38) ~[app.jar:0.0.1-SNAPSHOT]
	at com.gocardless.http.HttpClient.handleErrorResponse(HttpClient.java:168) ~[app.jar:0.0.1-SNAPSHOT]
	at com.gocardless.http.HttpClient.execute(HttpClient.java:153) ~[app.jar:0.0.1-SNAPSHOT]
	at com.gocardless.http.HttpClient.execute(HttpClient.java:81) ~[app.jar:0.0.1-SNAPSHOT]
	at com.gocardless.http.HttpClient$1.call(HttpClient.java:110) ~[app.jar:0.0.1-SNAPSHOT]
	at com.github.rholder.retry.AttemptTimeLimiters$NoAttemptTimeLimit.call(AttemptTimeLimiters.java:78) ~[app.jar:0.0.1-SNAPSHOT]
	at com.github.rholder.retry.Retryer.call(Retryer.java:160) ~[app.jar:0.0.1-SNAPSHOT]
	at com.gocardless.http.HttpClient.executeWithRetries(HttpClient.java:114) ~[app.jar:0.0.1-SNAPSHOT]
	at com.gocardless.http.IdempotentPostRequest.execute(IdempotentPostRequest.java:45) ~[app.jar:0.0.1-SNAPSHOT]
```

**Comment by matt-thomson** [comment]
Hi Jan,

`dddddd` is the correct format for the `branch_code` parameter.  You're right that we the exception message could be better here - I'll add that to our backlog.  In the meantime, you can view information about which field(s) are invalid by catching the exception and calling [`getErrors`](http://gocardless.github.io/gocardless-pro-java/com/gocardless/errors/GoCardlessApiException.html#getErrors()).

The likely cause is that we check that the bank details you supply pass a modulus check, and that the bank account is reachable for direct debits.

Hope that helps!

**Comment by matt-thomson** [comment]
Hi Jan,

I'm going to close this issue, but feel free to reopen if you're still having problems!

**Comment by matt-thomson** [comment]
Hi Jan,

I've just published v1.18.0 of this library, which includes more detailed messages in validation failed exceptions.  Hope this helps!

Matt

**Comment by bhargavcode** [comment]
Hello,
Anyone have a valid sandbox api for GoCardLess.
Actually i have created but its always throwing an exception and app going crashes.

Please help.
Getting this error.

 " The access token you've used is not a valid sandbox API access token
        at com.gocardless.errors.GoCardlessErrorMapper.toException(GoCardlessErrorMapper.java:18)"
